rfcs/000-third-party-software-installation.md

```diff
@@ -0,0 +1,109 @@
+The act of installing the software is done through the following additional steps:
+
+1. Running `v build.vsh` or some other specified build script in the root of the repo.
+2. This will create an executable, along with all other
+   relevant files (like templates, etc) in the `bin` directory of the repo. Ideally, the resources will be embedded in the executable to allow for only one executable to be enough.
+3. The executable will be added to `~/.v/bin` for the user to call from anywhere.
+   TODO.
+   An additional **suggestion** is to have the `~/.vmodules` directory be replaced in favour of `~/.v/modules`.
+
+If there is another script (other than `build.vsh`) that generates the executable
+
+A sample `build.vsh` script is:
+
+```text
+println('Removing old artifacts...')
+rmdir_all('bin') or {}
+println('Done removing "bin" directory.')
+
+println('\nCreating new output directory "bin"...')
+mkdir('bin') ?
+println('Done creating "bin" directory.')
+
+println('\nChecking if everything is formatted correctly...')
+execute_or_panic('v fmt -verify .')
+println('Done checking formatting.')
+
+println('\nCompiling and building executable...')
+execute_or_panic('v -prod . -o bin/klonol')
+println('Done compiling and placing executable in "bin".')
+```
+
+Taken from [klonol](https://github.com/hungrybluedev/klonol/blob/6e36e5676542aae40e0124de13055643484bb015/build.vsh).
+
+Users will have to add `~/.v/bin` to their `PATH` once.
+
+We also make use of the `bin` directory to keep the other files in the root of the repository out of the search path.
+
+# Drawbacks
+
+- Security concerns. Perhaps users should take some extra steps to install software from third-party sources.
+- It will confuse users between the `v install` and `v fetch` or `v obtain` commands.
+- Users need to tinker with their `PATH`.
+- Third-party projects with the same name will cause namespace collision.
+
+# Rationale and alternatives
+
+- The alternative is to clone each project separately, build them and add their `bin` folders to `PATH`.
+- It is preferable to have a simpler system.
+
+# Prior art
+
+Rust's `cargo` also allows users to install third-party software. Same for `npm`.
+
+`cargo` asks the user to update their `PATH`.
+
+# Unresolved questions
+
+- How do we handle the versions of third-party software?
+
+  See discussion in the Pull Request: [here](https://github.com/vlang/rfcs/pull/22#issuecomment-1075975185)
+
+- How do we allow for multiple versions?
+
+  See discussion in the Pull Request: [here](https://github.com/vlang/rfcs/pull/22#issuecomment-1075975185)
+
+- How do we handle namespace collisions?
+
+  See discussion in the Pull Request: [here](https://github.com/vlang/rfcs/pull/22#issuecomment-1075975185)
+
+- How do we allow for local environments? Like Python's Virtual Environment?
+
+  Probably needs to be discussed in a separate RFC.
+
+# Future possibilities
+
+We should keep the tentative location of `VHOME` customisable.
```

Comment:
What is `VHOME`?

Reply:
`VHOME` will be the `~/.v` directory containing the `~/.v/modules` and `~/.v/bin` directories as discussed above. This can be set via an environment variable just like `VMODULES`.

Reply:
@hungrybluedev as a side note, on Linux and BSD systems it should not default to `~/.v` but to `$XDG_DATA_HOME` or alike as per [XDG spec](https://specifications.freedesktop.org/basedir-spec/latest/ ) spec.

This will remove the necessity of having anything V-specific like VHOME (yeah, on Windows, Mac, Android etc. it's totally fine using XDG_DATA_HOME as well, so the fallback of setting it yourself if empty is fully cross-platform).